### PR TITLE
Modify GetFileURL Coordination Algorithm to Batch Files into Tasks by Size

### PR DESCRIPTION
**Pull request by FahadBSyed**

### CODE REVIEW

File: src/server/pfs/server/url_worker.go [hunk 1]
```diff
@@ -176,7 +175,6 @@ func (d *driver) processGetFileURLTask(ctx context.Context, task *GetFileURLTask
 }
 
 func serializePutFileURLTask(task *PutFileURLTask) (*types.Any, error) {
-	fmt.Printf("task: %+v\n", task)
```
Comment (brendoncarroll) [inline src/server/pfs/server/url_worker.go:179]: Not sure if you forgot this while debugging.  Remove or, if you want a log line, use our internal `log` package.

**Comment by brendoncarroll** [comment]
I'm not sure if we have logging in the task.Doer.  If we don't can you add some logging using our internal log package.
- every time we submit a task on the `inputChan`.  Total paths and total bytes would be good to include.
- Once all the tasks have been submitted and we are just waiting for the results.

File: src/server/pfs/server/url_worker.go (hunk 1 above)
Comment (FahadBSyed) [reply src/server/pfs/server/url_worker.go:179]: Yeah I think this is part of PFS-25, so I will remove that from there and this diff should vanish.

File: src/server/pfs/server/url.go [hunk 2]
```diff
@@ -186,7 +186,7 @@ func (d *driver) getFileURL(ctx context.Context, taskService task.Service, URL s
 		err := doer.Do(ctx, inputChan, func(_ int64, output *types.Any, err error) error { return err })
 		return errors.EnsureStack(err)
 	})
-	var bytesWritten int64
+	var size int64
```
Comment (brycemcanally) [inline src/server/pfs/server/url.go:189]: This isn't correct. Bytes written is separate from the sharding criteria. We should revert this back to bytes written and collect it independent of the size variable. Also, we should rename count to numObjects for consistency.

File: src/server/pfs/server/testing/server_test.go [hunk 3]
```diff
@@ -5385,21 +5385,23 @@ func TestPFS(suite *testing.T) {
 		for _, path := range paths {
 			require.NoError(t, env.PachClient.PutFile(commit, path, strings.NewReader(path)))
 		}
+		objC := dockertestenv.NewTestObjClient(ctx, t)
+		bucketURL := objC.BucketURL().String()
 		check := func() {
-			objC := dockertestenv.NewTestObjClient(ctx, t)
```
Comment (brycemcanally) [inline src/server/pfs/server/testing/server_test.go:5389]: Why was this change needed?

Comment (FahadBSyed) [reply src/server/pfs/server/testing/server_test.go:5389]: I wanted to add a case to test importing a directory and not individual files. My goal was to structure the code to reduce the amount of redunant lines, so I moved some of the steps out of check(). This way I could reset after testing individual file imports and test importing a directory.

File: src/server/pfs/server/url.go (hunk 2 above)
Comment (FahadBSyed) [reply src/server/pfs/server/url.go:189]: Good catch, size is getting reset, but bytesWritten shouldn't be.

File: src/server/pfs/server/url.go [hunk 4]
```diff
@@ -211,23 +211,24 @@ func (d *driver) getFileURL(ctx context.Context, taskService task.Service, URL s
 			}
 			return nil
 		}
-		var count int64
+		var numObjects, size int64
 		if err := src.Iterate(ctx, func(fi *pfs.FileInfo, file fileset.File) error {
 			if fi.FileType != pfs.FileType_FILE {
 				return nil
 			}
+			numObjects++
```
Comment (brycemcanally) [inline src/server/pfs/server/url.go:219]: I think the update should be after the check. Path ranges are inclusive / exclusive.

File: src/server/pfs/server/testing/server_test.go (hunk 3 above)
Comment (brycemcanally) [reply src/server/pfs/server/testing/server_test.go:5389]: The check function pattern in the server code is intended to ensure that the same access patterns work before and after finishing a commit. This sort of removes that since it now changes what is being checked against (files vs directory) before and after finishing the commit. I think we should just add the directory case to the check. So, it could be structured like: check files, clear object store, check directories.

**Review by brycemcanally — APPROVED** [review]
lgtm
